apm-agent-core/src/test/java/co/elastic/apm/agent/testutils/assertions/metrics/MetricsetEventJson.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.testutils.assertions.metrics;
+
+public class MetricsetEventJson {
+
+    public MetricsetJson metricset;
```

Comment:
can we use explicit mapping here (or make jackson throw an error when there is a field not mapped) ?

Reply:
See https://github.com/elastic/apm-agent-java/pull/2968#discussion_r1087746370: This will throw an error if any unmapped fields are found by default.